interfaces/apparmor/template_vars.go

```diff
@@ -35,6 +36,11 @@ func templateVariables(info *snap.Info, securityTag string) string {
 	fmt.Fprintf(&buf, "@{SNAP_REVISION}=\"%s\"\n", info.Revision)
 	fmt.Fprintf(&buf, "@{PROFILE_DBUS}=\"%s\"\n",
 		dbus.SafePath(securityTag))
-	fmt.Fprintf(&buf, "@{INSTALL_DIR}=\"/snap\"")
+
+	if dirs.SnapMountDir == "/snap" {
+		fmt.Fprintf(&buf, "@{INSTALL_DIR}=\"/snap\"")
+	} else {
+		fmt.Fprintf(&buf, "@{INSTALL_DIR}=\"{/snap,%s}\"", dirs.SnapMountDir)
+	}
```

Comment:
The changes look good. This comment describes why this is needed: https://github.com/snapcore/snapd/blob/master/cmd/snap/cmd_run.go#L658. I'm not sure why that comment is there though-- perhaps @zyga can comment?

It would be nice to add a test case for this in interfaces/apparmor/backend_test.go.

Reply:
This comment isn't entirely clear to me as it talks about classic snaps while I tested everything with strict snap sandbox.

Reply:
Ah, you're right. Plus, that is only for running under strace. Seems that this is possibly coming from getSnapInfo() (https://github.com/snapcore/snapd/blob/master/cmd/snap/cmd_run.go#L240), which snapRunApp() calls.

Reply:
The thing I don't understand is why apparmor needs to know about the /var/lib/snapd/snap path at all, we only ever use apparmor under confinement and when snaps are already bind mounted to /snap. I don't mind merging this but I think we are missing something.

Reply:
@zyga I uploaded strace output but don't know if it's useful:
https://forum.snapcraft.io/t/apparmor-issues-with-default-install-dir/4568/5